docs/apache-airflow-providers-amazon/executors/ecs-executor.rst

```diff
@@ -187,6 +187,13 @@ Then you can build your image by ``cd``-ing to the directory with the Dockerfile
    docker build -t my-airflow-image \
     --build-arg aws_default_region=YOUR_DEFAULT_REGION .
 
+For users on Apple Silicon, you must specify the arch using ``docker buildx``:
```

Comment:
Could you please update batch executor documentation as well. It is done here: https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-amazon/executors/general.rst?plain=1#L68. I wondered if we should not use this one in ECS executor as well? I guess that's why we moved this section to a separate file? @syedahsn @o-nikolas

Reply:
But thanks for adding this section! :D Very helpful for Mac users

Reply:
I was just digging all this up and got distracted before finishing it. Thanks @vincbeck!

I'm not sure why we haven't circled back to update ECS to use the generic doc imports. @syedahsn added them during the Batch project and perhaps we just haven't circled back?

Reply:
Yes, I'm pretty sure it was something that was on my todo list, but never got around to it. The ECS guide can be re-done to use the general document, similar to the Batch documentation.

Reply:
Thanks for clarifying everyone :) I modified ECS docs to now make use of `general.rst`, and modified `general.rst` to include Mac build instructions!

Reply:
It is not 100% correct, e.g. if it intend to be run under Graviton then it should be `--platform=linux/arm64`.

BTW It is better to build on the same architecture as on which intended to be launched

Reply:
So maybe instead of cover one case, better cover most of them with mention that is important to build and run under the same architecture. And also would be nice to have a link to the Docker documentation.

And keep example with a small clarification, that if you build on Apple Silicone and intend to run on X86_64 aka AMD64 then make sure that you use appropriate platform.

Reply:
Sorry I forgot to follow up on this!

I've now added a general comment noting that we must build and run under the same architecture. I added the mac example command, and also linked Docker documentation.